sycl/include/sycl/ext/oneapi/experimental/syclbin_kernel_bundle.hpp

```diff
@@ -0,0 +1,83 @@
+//==---- syclbin_kernel_bundle.hpp - SYCLBIN-based kernel_bundle tooling ---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/ext/oneapi/properties/properties.hpp>
+#include <sycl/kernel_bundle.hpp>
+
+#include <fstream>
+#include <string>
+
+#if __has_include(<filesystem>)
+#include <filesystem>
+#endif
+
+#if __has_include(<span>)
+#include <span>
+#endif
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+
+template <bundle_state State, typename PropertyListT = empty_properties_t>
+std::enable_if_t<State != bundle_state::ext_oneapi_source, kernel_bundle<State>>
+get_kernel_bundle(const context &Ctxt, const std::vector<device> &Devs,
+                  const sycl::span<char> &Bytes, PropertyListT = {}) {
+  std::vector<device> UniqueDevices =
+      sycl::detail::removeDuplicateDevices(Devs);
+
```

Comment:
Should we throw an `errc::invalid` exception if `Devs` is empty?

Reply:
Ahh, indeed. I found them in that func before and forgot to remove the comment:)

Reply:
Haha! Totally fair. Had me double-check too, which is good. 😄